src/pal/src/misc/stackstring.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+#include "pal/malloc.hpp"
+#include "pal/dbgmsg.h"
+
+SET_DEFAULT_DEBUG_CHANNEL(MISC);
+
+template <SIZE_T STACKCOUNT>
+class StackString
+{
+private:
+    WCHAR m_innerBuffer[STACKCOUNT + 1];
+    WCHAR * m_buffer;
+    SIZE_T m_count; // actual allocated count
+
+    void NullTerminate()
+    {
+        m_buffer[m_count] = W('\0');
+    }
+
+    void DeleteBuffer()
+    {
+        if (m_innerBuffer != m_buffer)
+            PAL_free(m_buffer);
+        else
+            m_buffer = NULL;
```

Comment:
You should set m_buffer to NULL unconditionally here (even for the case where m_innerBuffer != m_buffer).